### ISSUE TITLE
2.1.3: AddRedirect no longer works

### ISSUE THREAD

**Issue by ane-dwarf** [skybrud/Skybrud.Umbraco.Redirects#131]
#### Issue description

After updating to [newest release](https://github.com/skybrud/Skybrud.Umbraco.Redirects/releases/tag/v2.1.3) (v2.1.3) we are no longer able to create redirects using the following two lines of code:

```
var redirectDestination = new RedirectDestination(content.Id, content.Key, realtiveUrl, realtiveUrl, RedirectDestinationType.Content);

_redirectsService.AddRedirect(root.Id, path, redirectDestination, true, false, false);
```

Well, actually it does work if you set `DestinationQuery` and `DestinationFragment` to nullable in the database.
The `AddRedirect` overload with `AddRedirectOptions` works out-of-the-box:

```
var redirectDestination = new RedirectDestination(content.Id, content.Key, relativeUrl, relativeUrl, RedirectDestinationType.Content);

var addRedirectOptions = new AddRedirectOptions
{
    Destination = redirectDestination,
    IsPermanent = true,
    OriginalUrl = path,
    RootNodeId = root.Id,
    RootNodeKey = root.Key
};

_redirectsService.AddRedirect(addRedirectOptions);
```

#### Steps to reproduce the issue

Adding a redirect using [`AddRedirect(int rootNodeId, string url, RedirectDestination destionation, bool permanent, bool isRegex, bool forwardQueryString)`](https://github.com/skybrud/Skybrud.Umbraco.Redirects/blob/4614da271cc82763591fe526fc518530a141e7eb/src/Skybrud.Umbraco.Redirects/Models/RedirectsService.cs#L152) will cause an `Unable to insert redirect into the database`:

```
Cannot insert the value NULL into column 'DestinationQuery', table 'xx.dbo.SkybrudRedirects'; column does not allow nulls. INSERT fails.
```

#### What's the expected result?

Redirect should be created with empty values for `DestinationQuery` and `DestinationFragment`.

#### What's the actual result?

RedirectsService throws an error.

**Comment by abjerner** [comment]
Hi @ane-dwarf and thanks for reporting this 👍 

IIRC the columns (old and new) are not nullable as that caused some problems when generating the `WHERE` clauses from C#. I'm not sure whether this was only an issue in PetaPoco (Umbraco 7), or also NPoco (Umbraco 8 and 9). So making the columns nullable could lead to other issues.

I think I've missed setting the query string and fragment in the `AddRedirect(int rootNodeId, string url, RedirectDestination destionation, bool permanent, bool isRegex, bool forwardQueryString)` method. I don't recall it being used internally in the package any more, but if since it's public for others to use, it should of course work. I will look into that 😉

**Comment by NBM-Trifork** [comment]
Any news to this issue, I have the same in umbraco 9

**Comment by abjerner** [comment]
@NBM-Trifork I haven't had the time to look into this yet. But the `AddRedirect` that causes problems in V8 doesn't exist in the V9 package. Could you elaborate a bit about the issues that you're experiencing?

**Comment by abjerner** [comment]
@ane-dwarf This should now be fixed in the [**v2.0.4**](https://github.com/skybrud/Skybrud.Umbraco.Redirects/releases/tag/v2.1.4) release ;)

@NBM-Trifork I'm closing the issue for now as I'm not seeing the same problem in the V9 package. But if you have some more information, please do share, and I'll look into it.
